timetravel.py

```diff
@@ -1,166 +1,144 @@
+                "Spilaðu Kastljós",
                 "Spila Kiljuna frá í gær",
                 "Spila Tíufréttir frá því í fyrradag",
             )
         )
     )
 
+
 # This module wants to handle parse trees for queries
 HANDLE_TREE = True
 
 # The grammar nonterminals this module wants to handle
-QUERY_NONTERMINALS = {"QTimeTravel", "QStartOver"}
+QUERY_NONTERMINALS = {"QTVTimeTravel"}
 
 # The context-free grammar for the queries recognized by this plug-in module
 GRAMMAR = """
 
-Query →
-    QTimeTravel | QStartOver
+Query → QTVTimeTravel
  
-QTimeTravel → QTimeTravelQuery '?'?
+QTVTimeTravel → QTVTimeTravelQuery '?'?
 
-QStartOver → QStartOverQuery '?'?
+QTVTimeTravelQuery →
+    QTVStartOver
+    | QTVTimeTravelKeyword QTVTimeTravelProgram QTVTimeTravelSinceWhen?
 
-QTimeTravelQuery →
-    QTimeTravelKeyword QTimeTravelProgram QTimeTravelWhen?
-
-QStartOverQuery →
+QTVStartOver →
     "byrja" "upp" "á" "nýtt"
-    | "byrja" "byrjun"
+    | "byrja" "frá" "byrjun"
+    | "aftur" "á" "byrjun"
 
-QTimeTravelCourtesy →
-    "getur" 
-    | "getur" "þú" 
+QTVTimeTravelCourtesy →
+    "getur" "þú"?
     | "geturðu" 
 
-QTimeTravelKeyword →
-    "spila" | "spilar" | "spilaðu" | "spilað"  | "spilaði"
+QTVTimeTravelKeyword →
+    "spila"
+    | "spilar"
+    | "spilaðu"
+    | "spilað"
+    | "spilaði"
 
-QTimeTravelAtviksord →
-    "í" | "á" | "við" | "þau" | "þú"
+# QTVTimeTravelAtviksord →
+#     "í" | "á" | "við" | "þau" | "þú"
 
-QTimeTravelProgram →
+QTVTimeTravelProgram →
     Nl
 
-QTimeTravelSince →
+QTVTimeTravelSinceWhen →
+    QTVTimeTravelSince? QTVTimeTravelWhen
+
+QTVTimeTravelSince →
     "síðan" 
     | "frá" 
     | "frá" "það" 
     | "frá" "því"
 
-QTimeTravelWhen →
-    QTimeTravelToday | QTimeTravelYesterday | QTimeTravelDayBeforeYesterday
+QTVTimeTravelWhen →
+    QTVTimeTravelToday
+    | QTVTimeTravelYesterday
+    | QTVTimeTravelDayBeforeYesterday
 
-QTimeTravelToday →
-    "í" "dag"
+QTVTimeTravelToday →
+    "í" "dag" | "í_kvöld"
 
-QTimeTravelYesterday →
-    "í_gær" | "geir" | "hér"
+QTVTimeTravelYesterday →
+    "í_gær"
+    # | "geir"
+    # | "hér"
 
-QTimeTravelDayBeforeYesterday →
+QTVTimeTravelDayBeforeYesterday →
   "í_fyrradag"
 
 """
 
 
-def QTimeTravelQuery(node, params, result):
-    result.qtype = _TIMETRAVEL_QTYPE
+def QTVTimeTravelQuery(node, params, result):
+    result.qtype = _TV_TIMETRAVEL_QTYPE
+
 
-def QStartOverQuery(node, params, result):
-    result.qtype = _STARTOVER_QTYPE
+def QTVStartOver(node, params, result):
+    result["qkey"] = _STARTOVER_CMD
 
-def QTimeTravelProgram(node, params, result):
-    nl = NounPhrase(result._nominative)
-    try:
-        result["program-angr"] = "{nl:ángr}".format(nl=nl)
-        result["program-nf"] = "{nl:nf}".format(nl=nl)
-    except:
-        result["program-angr"] = str(nl)
-        result["program-nf"] = ""
 
-    print("Nl:", nl)
+def QTVTimeTravelProgram(node: Node, params: ParamList, result: Result):
```

Comment:
Þetta virkar fyrir t.d. "Fréttir" en ekki fyrir "Gísla Martein" né "Silfur Egils" sem innihalda mannanöfn.

```{
   "q_raw": "spila fréttir frá því í gær",
   "q": "Spila fréttir Frá því í gær.",
   "num_sent": 1,
   "num_parsed_sent": 1,
   "answer": "TIMETRAVEL;fréttir;yesterday",
   "response": {
      "answer": "TIMETRAVEL;fréttir;yesterday"
   },
   "qtype": "TV-Timetravel",
   "valid": true
}
Query is: 'Spila Gísla Martein Frá því í gær?'
Unable to parse query, error E_NO_PARSE_TREES
Query is: 'Spila silfur Egils Frá því í gær?'
Unable to parse query, error E_NO_PARSE_TREES```

Reply:
Hvaða version af bíntokenizernum (greynirpackage) og tokenizernum ertu að nota? þetta virðist amk virka hjá mér :thinking:

Reply:
vorum búin að laga 'Frá' villuna í sumar er ég best veit